In [1]:
import pandas as pd
from collections import Counter
import matplotlib.pyplot as plt
import numpy as np
import statistics 

df= pd.read_csv("/home/danielgarcia/baby_shark_dooo_dooo/input/attacks.csv", encoding='latin-1')

In [2]:
df=df.rename(columns={'Species ': 'Species','Sex ': 'Sex','Fatal (Y/N)': 'Fatal'},)

In [3]:
df=df.reindex(['Species','Type','Activity','Injury','Fatal','Country','Area','Location','Sex','Age','Year'], axis='columns')
df=df.dropna()
df = df.round(decimals=0).astype(object)

In [4]:
df['Age']=df['Age'].str.extract('(\d+)', expand=False)

In [5]:
df['Year']=df['Year'].dropna()
df['Age'] = pd.to_numeric(df['Age'])
df['Year'] = pd.to_numeric(df['Year'])
df['Age'].unique
df.loc[df['Year'] ==0, 'Year'] = 2018 - df['Age']

# ¿Cuan peligrosos son los ataques de tiburones hacia los seres humanos?

In [6]:
a=df['Year'].min() ,df['Year'].max()
df.drop(df.loc[df['Type']=='Invalid'].index, inplace=True)
#print (f"El primer año con ataque registrado es {int(a[0])} y el ultimo es {int(a[1])}.")

El primer año con ataque registrado es 1868 y el ultimo es 2018.


### El DataSet  nos muestra informacion recopilada entre 1868 y 2018, esto son 150 años de informacion recopilada a largo de todo el globo terraqueo.

In [7]:
df['Year'].value_counts()
df['Year'].min()
Rango_ataques=int(df['Year'].max()-df['Year'].min())


150

#### Puesto que queremos ver si realmente los tiburones son tan peligrosos para los humanos analizaremos los datos relacionados con el genero, la edad y si el ataque fue definitivo o se quedo en un susto.                                                                         El primer dato calculado es la edad media de los participantes en el estudio, observamos que son 28 años, la mediana con 25 y la desviacion tipica es de 13.  Los valores extremos son 84 años y 5 años, la edad dividida por cuartiles nos indican grandes varianzas, mientras que el cuartil 0,5 se situa en los 25 años (coincidiendo con la mediana y siendo un poco menor que la media) , el 3 cuantil nos situa en los 36 años vemos un tremendo gap hasta el final con los 84 años. 

In [8]:
Age_Mean=df["Age"].mean()
Age_Median=df["Age"].median()
Age_std=df["Age"].std()
#print(f"La edad media de los que han sufrido un ataque es de {int(Age_Mean)}, el valor mas repetido, esto es, la mediana con {int(Age_Median)} y la desviacion es {int(Age_std)}.")
#print (f"La edad maxima de los atacados es de {int(df['Age'].max())} y la minima de {int(df['Age'].min())}")
a=df['Age'].quantile([0.25,0.5,0.75,1]).quantile


### El numero total de voluntarios que se ofrecieron para realizar el estudio es de 1844.Si empezamos a aplicar filtros veremos informacion mas precisa, yo he decidido utilizar filtros demograficos para ver si hay patrones que se repiten, tambien desglose la informacion en base a si la victima era una victima mortal o no puesto que estaba interesado en ver informacion relacionada. 

In [9]:
def len_gen(df):
    return len(df[df['Sex'] == 'F']),len(df[df['Sex'] == 'M'])
#len_gen(df)
#print(f"EL numero de mujeres registradas es de {len_gen(df)[0]} y el de hombres es de {len_gen(df)[1]}.")

(246, 1607)

In [10]:
def len_fa(df):
    return len(df[df['Fatal'] == 'N']),len(df[df['Fatal'] == 'Y'])
#print(len_fa(df))
#print(f"El numero total de supervivientes es de {len_fa(df)[0]} y de fallecidos es de {len_fa(df)[1]} .")

### Ciñendonos al genero podemos observar que los hombres suelen mas aventureros o insconcientes (dependiendo de como queramos denominarlo) pero frente a la desviacion media femenina situada en los 18 años la de los hombres es de tan solo 13 años, huelga decir que esto fue de las victimas mortales. Basandonos en los supervivientes vemos que la de los hombres supervivientes es practicamente igual mientras que la de la mujer es incluso menor, posicionandose en los 16 años, poca diferencia pero que puede ser crucial para el desarrollo de uno (ya sea personalmente o como comida para tiburones). Las victimas letales mas longevas se situan en 84 (por lo tanto el rango maximo de la muestra no lo consiguio) masculino y 77 femenina, los menores  son 5 y 6 (y por lo tanto la muestra menor del estudio sobrevivio). No obstante los menores fallecidos ambos poseian 7 años

In [11]:
a=df.loc[(df['Fatal']=='N') & (df['Sex']=='M' )& (df['Age']),['Fatal','Sex','Age']]
b=df.loc[(df['Fatal']=='N') & (df['Sex']=='F' )& (df['Age']),['Fatal','Sex','Age']]
c=df.loc[(df['Fatal']=='Y') & (df['Sex']=='M' )& (df['Age']),['Fatal','Sex','Age']]
d=df.loc[(df['Fatal']=='Y') & (df['Sex']=='F' )& (df['Age']),['Fatal','Sex','Age']]
#'''print (int(c.median()[0]))
#print(c.std()[0])
#print (int(d.median()[0]))
#print(d.std()[0])
#print (int(c.median()[0]))
#print(c.std()[0])
#print (int(d.median()[0]))
#print(d.std()[0])
#print (f"La edad maxima registrada del superviviente masculino es de: {int(a['Age'].max())}\nLa de las mujeres supervivientes es de: {int(b['Age'].max())}\nPor otro lado la del fallecido masculino mas adulto de: {int(c['Age'].max())} y la de la mujer fallecida es de: {int(d['Age'].max())}")


#print (f"La edad minima registrada del superviviente masculino es de: {int(a['Age'].min())}\nLa edad minima de las mujeres supervivientes es de: {int(b['Age'].min())}\nPor otro lado la del fallecido masculino mas joven de: {int(c['Age'].min())} y la de la mujer fallecida es de: {int(d['Age'].min())}\n\n")



#print (f'La edad media de supervivientes masculinos es de: {a.mean()[0]}\nLa de las mujeres supervivientes es de : {b.mean()[0]}\nLa edad media de los fallecidos masculinos era de: {c.mean()[0]}\nLa de las mujeres fallecidas de: {d.mean()[0]}')'''

'print (int(c.median()[0]))\nprint(c.std()[0])\nprint (int(d.median()[0]))\nprint(d.std()[0])\nprint (int(c.median()[0]))\nprint(c.std()[0])\nprint (int(d.median()[0]))\nprint(d.std()[0])\nprint (f"La edad maxima registrada del superviviente masculino es de: {int(a[\'Age\'].max())}\nLa de las mujeres supervivientes es de: {int(b[\'Age\'].max())}\nPor otro lado la del fallecido masculino mas adulto de: {int(c[\'Age\'].max())} y la de la mujer fallecida es de: {int(d[\'Age\'].max())}")\n\n\nprint (f"La edad minima registrada del superviviente masculino es de: {int(a[\'Age\'].min())}\nLa edad minima de las mujeres supervivientes es de: {int(b[\'Age\'].min())}\nPor otro lado la del fallecido masculino mas joven de: {int(c[\'Age\'].min())} y la de la mujer fallecida es de: {int(d[\'Age\'].min())}\n\n")\n\n\n\nprint (f\'La edad media de supervivientes masculinos es de: {a.mean()[0]}\nLa de las mujeres supervivientes es de : {b.mean()[0]}\nLa edad media de los fallecidos masculinos era de: {c

###  Mientras que la edad media de los supervivientes masculinos es de 28.3 años la de las mujeres es de 27.54, con los fallecimientos se invierte y es la mujer la que posee mas edad(30años) vs 27 del hombre

In [12]:
pl=df.loc[(df['Fatal']=='N') & (df['Sex']=='M' ),['Fatal','Sex']]
(np.sum(pl, axis=1).tolist)
#len(pl)
#print (f"EL numero de hombres vivos es de {len(pl)}.")

1361

In [13]:
lmu=df.loc[(df['Fatal']=='N') & (df['Sex']=='F' ),['Fatal','Sex']]
np.sum(lmu, axis=1).tolist
len(lmu)
#print (f"EL numero de mujeres vivas es de {len(lmu)}")


204

In [20]:
depm=df.loc[(df['Fatal']=='Y') & (df['Sex']=='M' ),['Fatal','Sex']]
(np.sum(depm, axis=1).tolist)
len(depm)


241

In [19]:
depmu=df.loc[(df['Fatal']=='Y') & (df['Sex']=='F' ),['Fatal','Sex']]
np.sum(depmu, axis=1).tolist
len(depmu)


42

### El total de supervivientes por genero es de 1.361(73%) vs 204(11%)  de la muestra total.  El numero de hombres fallecidos es de 240 siendo esto un 13% frente a un 2% (42 victimas ) femeninas.

In [16]:
Leth= (len(df[df['Fatal'] == 'Y'])*100/len(df[df['Fatal'] == 'N']))
Leth
#La letalidad de los ataques mostrados es de 18.22

18.071519795657725

### Si comparamos los ataques letales frente a los no letales vemos un 18% de posibilidades de morir bajo las fauces de un tiburon, cifras comedidas.

In [17]:
SuPr=df.loc[(df['Fatal']=='N') & (df['Type']=='Provoked' ),['Fatal','Type']]
LePr=df.loc[(df['Fatal']=='Y') & (df['Type']=='Provoked' ),['Fatal','Type']]
SuPr.size, LePr.size, df['Type'].size
tnp=(df['Type'].size)- (SuPr.size+LePr.size)
#print (f'El total de casos provocados es de {SuPr.size+LePr.size} y el de casos no provocados: {tnp}, de los cuales solo {LePr.size} han sido provocados y mortales.')
#ataques provocados y letalidad 

In [18]:
Surfinga=df.loc[(df['Fatal']=='N') & (df['Type']=='Provoked' ) &(df['Sex']=='M'),['Fatal','Type','Sex']].shape
b=df.loc[(df['Fatal']=='N') & (df['Type']=='Provoked' ) &(df['Sex']=='F'),['Fatal','Type','Sex']].shape
c=df.loc[(df['Fatal']=='N') & (df['Type']=='Unprovoked' ) &(df['Sex']=='M'),['Fatal','Type','Sex']].shape
d=df.loc[(df['Fatal']=='N') & (df['Type']=='Unprovoked' ) &(df['Sex']=='F'),['Fatal','Type','Sex']].shape
#print(f"El numero de ataques provocados por hombres sin victimas letales es de: {a[0]} y el de mujeres es de {b[0]}\nEl numero de ataques sin provocar y sin resultado mortal a hombres es de {c[0]} y el de mujeres {d[0]}")

### Observamos que la mayoria de ataques no fueron provocados por lo que si uno sigue el sentido comun y no molesta a los escualos si estos aparecen a tu alrededor obtendra mayores oportunidades para sobrevivir. No obstante, tan solo 4 muestras murieron al provocar al tiburon por lo que si decides hacerlo ten cuidado. 
### Para acabar observamos que la mayoria de ataques son sin provocar y sin resultado mortal (1153M y 185F)

### Con los numeros vistos, la escasa cifra de victimas mortales, el tiempo utilizado en la muestra siendo superior a un siglo, podemos asegurar que es mas factible morir por el baby shark que ante el verdadero ataque de un tiburon.

https://www.youtube.com/watch?v=5w8QEWA8wGM